### ISSUE TITLE
when purchasing multiples of an item, Amplitude does not indicate number purchased

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#9462]
Most purchasable items now let you buy multiples at once by selecting the number from a drop down or typing it. When you do a purchase like that, our Amplitude data does not say how many items were bought. For example, this is the record from me buying 18 potions:
```
acquireMethod: Gems
category: behavior
gemCost: 2
itemKey: Thunderstorm
itemName: Thunderstorm Hatching Potion
itemType: Market
```

That data should also contain the number of items purchased. E.g.:
```
amountPurchased: 18
```

This should occur for items bought with either gems or gold. I.e., any current code that sends purchase data to Amplitude should be modified to include the amount purchased (including the case where just 1 item is purchased).

**Comment by librarianmage** [comment]
@Alys should the key be `amountPurchased` or `quantity`?

Either way, I can work on this!

**Comment by Alys** [comment]
@MathWhiz Thank you! Let's make it `quantityPurchased`. I do like quantity better than amount, but I want to use a double word to assist with grepping for the term. Searching the code base for merely `quantity` would be stressful. :)

**Comment by librarianmage** [comment]
Ah, yeah.

**Comment by librarianmage** [comment]
Should it also record quantity for gear? It should always be 1.

**Comment by librarianmage** [comment]
Also, there is this

https://github.com/HabitRPG/habitica/blob/7fe25049063aae6de581dfac9f2fffd65cd9ef37/website/common/script/ops/buy.js#L27-L48

Which means that each individual purchase is recorded on Amplitude.

**Comment by negue** [comment]
I think this should be post-poned until the `buy/purchase`-methods merged / refactored

**Comment by Alys** [comment]
No, this shouldn't be postponed. We need this information for
troubleshooting bug reports related to gem purchases. I expect it's also
important for tracking the popularity of gem-purchaeable items; for
example, there's a large difference between one purchase of 1 potion and
one purchase of 18 potions.

MathWhiz, for equipment I don't care if quantityPurchased is included or
not. If it's easier or more consistent to include it with a value of 1,
then that's fine. Or leave it out if that's simpler.

**Comment by Alys** [comment]
@MathWhiz I've just got online and have looked at the code you posted - thank you  for finding that! 

Hatching Potions aren't included in that else-if block and my tests show that when I buy multiple Hatching Potions at once, Amplitude records only a single event with no indication of the quantity purchased, so Hatching Potions at least need to be fixed.

I agree that the code you posted does look like it handles the other purchase types correctly, but worryingly, when I purchased four quest scrolls in a single purchase just now, Amplitude recorded only one event, with no indication of quantity purchased. A second purchase of another four quests had the same effect. In each case, I did end up with four quest scrolls. Something seems broken there but I haven't had time to look at the code that does the actual purchases and sends the Amplitude data. Maybe quest scroll purchases aren't actually done in the code around that else-if block.

Buying gems with gold results in the correct number of events in Amplitude (one event for each gem purchased, as indicated in the code you posted).

If you want to initially do a PR that fixes only the Hatching Potions case, that would be fine with me. It might be a fast fix and it's one of the most important cases for this issue.

Quest scrolls are also important so if you can work out what's going wrong there, that would be awesome, but feel free to do that in a separate PR if you prefer, or leave it for later if you run out of time. 

Basically, any improvement you can make to this situation will be helpful for troubleshooting bug reports!

**Comment by librarianmage** [comment]
@Alys It appears to be that the code to buy quests is here:

https://github.com/HabitRPG/habitica/blob/7fe25049063aae6de581dfac9f2fffd65cd9ef37/website/common/script/ops/purchase.js#L109-L162
